src/nccl_ofi_sendrecv.cpp

```diff
@@ -2720,6 +2720,28 @@ int nccl_net_ofi_sendrecv_init(const char *provider_filter,
 		goto error;
 	}
 
+	struct fi_info *iter;
+	iter = provider_list;
+	while (iter != NULL) {
+		if (FI_VERSION_LT(fi_version(), FI_VERSION(2, 2)) && 
```

Comment:
All the info objects in provider_list will have the same prov_name at this point, so it would be much more efficient in the common case to write this as:
```C
/* big comment here */
if (FI_VERSION_LT(fi_version(), FI_VERSION(2,2)) && strcmp(iter->fabric_attr->prov_name, "tcp") == 0) {
  struct fi_info *iter = provider_list;
  while (iter != NULL)
     iter->caps |= FI_RMA;
     iter = iter->next;
  }
}
```